artifacts/reader.py

```diff
@@ -117,62 +165,53 @@ def _ReadArtifactDefinition(self, yaml_definition):
       if source_type:
         source_type.conditions = source.get(u'conditions', [])
         source_type.returned_types = source.get(u'returned_types', [])
-        self._ReadSupportedOS(yaml_definition, source_type, name)
+        self._ReadSupportedOS(source, source_type, name)
+        for supported_os in source_type.supported_os:
+          if supported_os not in artifact_object.supported_os:
```

Comment:
Ok, I will just code this up to have a validation error if they don't match rather than just forcing the addition.